### PR TITLE
test: test requires keyword - v1

### PR DESCRIPTION
**Pull request by jasonish**
Also includes readme update for files to exists as a requirement.

### CODE REVIEW

**Comment by victorjulien** [comment]
Can you add a test with the example rules I shared in the previous PR?

File: tests/requires/test.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        min-version: 8
+      count: 1
+      match:
+        event_type: alert
+  - filter:
+      requires:
+        min-version: 8
+      count: 1
+      match:
+        alert.signature_id: 8
```
Comment (inashivb) [inline tests/requires/test.yaml:21]: These could be one filter. Same for below filters. Why were they written separately?

File: tests/requires/test.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  # Check that for Suricata >= 8.0.0 we have one alert and that alert is sid 8.
```
Comment (inashivb) [inline tests/requires/test.yaml:23]: Typo in comment.

File: tests/requires/test.yaml (hunk 1 above)
Comment (jasonish) [reply tests/requires/test.yaml:21]: I don't think so. We want to confirm 2 things:
- There is only 1 alert, regardless of SID
- That alert is SID 8

If these checks were to be combined, we'd be testing that there is 1 or more alerts, but only one has SID 8. But there could still be more alerts with other SIDs.

Comment (inashivb) [reply tests/requires/test.yaml:21]: ah. ok. thanks for explaining!

**Comment by jasonish** [comment]
> Can you add a test with the example rules I shared in the previous PR?

Done.

**Review by victorjulien — APPROVED** [review]

File: tests/requires/test.rules [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+# Rule for Suricata >= 7.0.3 but less than 8... Or >= 8.0.1
+alert http any any -> any any (content:"uid=0"; requires: version >= 7.0.3 < 8 | >= 8.0.1; sid:9; rev:1;)
+
+# Rule for Suricata >= 8.
+alert http any any -> any any (msg:"TEST Suricata >= 8"; content:"uid=0"; requires: version >= 8.0.0; sid:8; rev:1;)
+
+# These rules have something invalid about them, but do follow the general rule
+# structure, so should be eliminated by the requires statement.
+alert vxlan any any -> any any (requires: version >= 10; sid:1;)
```
Comment (catenacyber) [inline tests/requires/test.rules:12]: I would put version 42 rather than 10 which is within our lifetimes I hope

File: tests/requires/test.rules [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Rule for Suricata >= 7 and < 8.
+alert http any any -> any any (msg:"TEST Suricata >= 7 and < 8"; content:"uid=0"; requires: version >= 7 < 8; sid:7; rev:1;)
+
+# Rule for Suricata >= 7.0.3 but less than 8... Or >= 8.0.1
+alert http any any -> any any (content:"uid=0"; requires: version >= 7.0.3 < 8 | >= 8.0.1; sid:9; rev:1;)
```
Comment (catenacyber) [inline tests/requires/test.rules:5]: Wow we allow complex "or" !

File: tests/requires/test.rules [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+# Rule for Suricata >= 8.
+alert http any any -> any any (msg:"TEST Suricata >= 8"; content:"uid=0"; requires: version >= 8.0.0; sid:8; rev:1;)
+
+# These rules have something invalid about them, but do follow the general rule
+# structure, so should be eliminated by the requires statement.
+alert vxlan any any -> any any (requires: version >= 10; sid:1;)
+alert udp any any -> any any (vxlan_vni:10; requires: version >= 10; sid:2;)
+alert http any any => any any (requires: version >= 10; sid:3;)
+alert tcp any any -> any any (frame:smtp.not_supported; requires: version >= 10; sid:4;)
```
Comment (catenacyber) [inline tests/requires/test.rules:15]: Could we also test `requires: version >= 10, superpower: notyet` ? (superpower being an unknown key yet for the `requires` keyword

**Comment by victorjulien** [comment]
Merged in #1558, thanks!

**Pull request closed without merging**